### DISCUSSION TITLE
continuous output flag on regex find component doesn't work

### DISCUSSION THREAD

**Discussion by seannybgoode** [FakeFishGames/Barotrauma#12545] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Regardless of whether or not the 'continuous output' flag is set on a regex component, the component will continue to produce the last output.

### Reproduction steps

Create a regex component signal with text input.
Set 'continuous output' flag false
Connect regex component to text output
Regex component will continue to produce output indefinitely once triggered.

### Bug prevalence

Just once

### Version

v1.0.13.2

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
This is how the "continuous output" setting is supposed to work. It'll keep sending the last output indefinitely.

**Comment by seannybgoode** [comment]
> This is how the "continuous output" setting is supposed to work. It'll keep sending the last output indefinitely.

I think you misread the steps in the issue. The behavior of the component is exactly the same regardless of whether or not the flag is set. Surely that's not intended behavior?

**Comment by Regalis11** [comment]
@seannybgoode Yeah, looks like I read this too hastily! I think I thought you assumed "continuous output" would not keep sending the previous output, but the output depending on what's currently being received (i.e. false output if it's receiving nothing). But if the component does always output a signal, regardless if it's receiving an input, something's definitely bugged here.

**Comment by seannybgoode** [comment]
Yeah that's what I'd expect as well. Double checking with a terminal -> regex -> text output, it looks like the issue is confirmed here. The regex component will output the last value received regardless of the flag. Thanks for taking a second look at this. I had to wire an elaborate reset circuit to work around this issue!

**Comment by Regalis11** [comment]
I gave this a proper test now, but I still could not reproduce the issue.

I have a suspicion you might be misunderstanding how the "set_text" input of the text display works: if you send a signal to it, that value will stay on the display indefinitely unless something else is received. So could it be the component isn't actually sending a signal when it's "continuous output" is disabled, the previous value just stays on the display?

...And on a sidenote, the wiring debugger we've got coming up in the next update (and that's currently up in the unstable version) should hopefully make diagnosing stuff like this much easier. We should've implemented it sooner!

**Reply by dossy** [reply]
Try using a Terminal instead of a Text Display.  Connect the SIGNAL_OUT from the RegExp Find Component to the SIGNAL_IN of the Terminal component, and regardless of the state of the "Continous output" setting on the RegExp Find Component, you'll see either the "Output" or the "False output" values of the RegEx Find Component spamming the Terminal.

The expected behavior is that if "Continous output" is unchecked, then the RegEx Find Component should only emit an output when the input changes.

Oh!  Haha, I have a Memory Component as the SIGNAL_IN to the RegEx Find Component, which is continuously feeding new inputs to the RegEx Find Component, so regardless of the "Continuous output" setting on the RegEx Find Component, it's going to continuously stream output in response to the continuous stream of inputs from the Memory Component.

Looks like the RegEx Find Component works as designed, and it's a matter of making sure that you don't wire up an input to it that is continuously outputting a signal.

Damn.

**Comment by seannybgoode** [comment]
@Regalis11 - I see what you're saying. That it'll output 1, then if no new input received it'll output nothing. Shouldn't it output a false signal? In lieu of a way to handle a null signal that is. In the case of a text display, the null output doesn't trigger a change. In the case of an AND component, the null output doesn't trigger a change either. 

Maybe I can suggest a more graceful way to handle it would be to leave the current behavior intact, and give the user a field in the regex component, so they can choose the null output - the signal that is output in lieu of input. That way we wouldn't break any current sub designs that rely on the null output, and the user could specify the behavior they want. 

Regardless of where you land, thanks for taking a look at this one. I was really scratching my head on it this week.
